lib/shopify_api/plugs/webhook_plug.ex

```diff
@@ -0,0 +1,198 @@
+defmodule ShopifyAPI.WebhookPlug do
+  @moduledoc """
+  A Plug to handle incoming webhooks from Shopify.
+
+  This plug requires access to the unparsed request body in order to perform HMAC verification.
+  As a result, it should be installed outside of your Phoenix router, and before any JSON parsing.
+  In most Phoenix apps, this should be somewhere inside of your Endpoint module (`lib/your_app_web/endpoint.ex`).
+
+  ## Options
+
+  There are two required and one optional bit of configuration for this Plug:
+
+  - `:prefix` - the request path for webhooks. Must be valid Plug.Router syntax.
+  - `:callback` - a {mod, fun, args} tuple to be invoked for each verified webhook received.
+  - `:app_name` (opt) - a default Shopify App name string to be used if one isn't present in the webhook request.
+
+  The provided `callback` is called from the plug itself, so take care to limit the work done in the callback itself.
+  If possible, enqueue a background job or otherwise handoff any further work to another asynchronous component.
+  This helps reduce the likelihood of slow response times between ourselves and Shopify's webhook backend.
+
+  ## Shopify App Routing
+
+  Shopify does not send the name of an app receiving the webhook along with the payload.
+  As such, there's two ways to make sure that requests are parsed and verified correctly:
+
+  1. Append your Shopify App's name to the Webhook URL configured in Shopify.
+     e.g. if your app is 'reporting-tool', make your webhook url end in '/webhooks/reporting-tool'.
+
+  2. Specify a default `app_name` as configuration when installing the plug (see below).
+
+  ## Example Installations
+
+      # No default app provided -- make sure to append the app name to webhook URLs in Shopify.
+      plug ShopifyAPI.WebhookPlug,
+           prefix: "/shopify/webhooks/",
+           callback: {WebhookHandler, :handle_webhook, []}
+
+      # Default app provided -- if app name is not present in URL, defaults to "reporting-tool".
+      plug ShopifyAPI.WebhookPlug,
+           prefix: "/webhook",
+           app_name: "reporting-tool",
+           callback: {WebhookHandler, :handle_webhook, []}
```

Comment:
I'm a little uncertain about this potential overconfiguration, but it does seem better than what we have now.